metadoc-cli/src/main/scala/metadoc/cli/TermDisplay.scala

```diff
@@ -0,0 +1,510 @@
+/**
+  * This code is copy/pasted from (Apache 2 licence)
+  * https://github.com/alexarchambault/coursier/blob/51fefe5c29d95752ce487f60d333b1f8a91dd1b0/cache/src/main/scala/coursier/TermDisplay.scala
+  *
+  * which in turn was copy/pasted from (MIT licence)
+  * https://github.com/lihaoyi/Ammonite/blob/10854e3b8b454a74198058ba258734a17af32023/terminal/src/main/scala/ammonite/terminal/Utils.scala
+  */
+package metadoc.cli
+
+import java.io.File
+import java.io.Writer
+import java.sql.Timestamp
+import java.util.concurrent._
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Try
+
+object Terminal {
+
+  private lazy val pathedTput =
+    if (new File("/usr/bin/tput").exists()) "/usr/bin/tput" else "tput"
+
+  def consoleDim(s: String): Option[Int] =
+    if (new File("/dev/tty").exists()) {
+      import sys.process._
+      val nullLog = new ProcessLogger {
+        def out(s: => String): Unit = {}
+        def err(s: => String): Unit = {}
+        def buffer[T](f: => T): T = f
+      }
+      Try(
+        Process(Seq("bash", "-c", s"$pathedTput $s 2> /dev/tty"))
+          .!!(nullLog)
+          .trim
+          .toInt
+      ).toOption
+    } else
+      None
+
+  implicit class Ansi(val output: Writer) extends AnyVal {
+    private def control(n: Int, c: Char) = output.write(s"\033[" + n + c)
+
+    /**
+      * Move up `n` squares
+      */
+    def up(n: Int): Unit = if (n > 0) control(n, 'A')
+
+    /**
+      * Move down `n` squares
+      */
+    def down(n: Int): Unit = if (n > 0) control(n, 'B')
+
+    /**
+      * Move left `n` squares
+      */
+    def left(n: Int): Unit = if (n > 0) control(n, 'D')
+
+    /**
+      * Clear the current line
+      *
+      * n=0: clear from cursor to end of line
+      * n=1: clear from cursor to start of line
+      * n=2: clear entire line
+      */
+    def clearLine(n: Int): Unit = control(n, 'K')
+  }
+
+}
+
+object TermDisplay {
+  def defaultFallbackMode: Boolean = {
+    val env0 = sys.env.get("COURSIER_PROGRESS").map(_.toLowerCase).collect {
```

Comment:
Maybe `METADOC_PROGRESS` or remove? ... or maybe it doesn't matter.

Reply:
Fixed.